service/history/shard/controller_impl.go

```diff
@@ -168,19 +178,10 @@ func (c *ControllerImpl) Stop() {
 	c.logger.Info("", tag.LifeCycleStopped)
 }
 
-// PrepareToStop starts the graceful shutdown process for controller
-func (c *ControllerImpl) PrepareToStop() {
```

Comment:
If PrepareToStop is going away, where will our clean shard shutdown code go (where we want to snapshot as many dirty MS objects as we can out of our cache to speed up the subequent shard reload)?